### PR TITLE
Register the runtime assertion handler for PsySH

### PR DESCRIPTION
**Pull request by damiankloip**
This is just getting ready for when https://www.drupal.org/node/2536560 is committed into 8.0.x . Once that is in, this will work, then we have a tiny patch that will actually fix the problem consistently by registering that handler in drush php : https://www.drupal.org/node/2550641 .

So we need to hold off until the first issue in done at least.

### CODE REVIEW

File: commands/core/cli.drush.inc [hunk 1]
```diff
@@ -20,10 +20,13 @@ function drush_cli_core_cli() {
   $shell = new Psy\Shell();
 
   if (drush_drupal_major_version() >= 8) {
+    // Register the assertion handler so exceptions are thrown instead of errors
+    // being triggered. The plays nicer with PsySH.
```
Comment (weitzman) [inline commands/core/cli.drush.inc]: typo. should be This.

Comment (damiankloip) [inline commands/core/cli.drush.inc]: Thanks. fixed!

**Comment by damiankloip** [comment]
https://www.drupal.org/node/2536560 is now in, just need https://www.drupal.org/node/2550641 now.

**Comment by weitzman** [comment]
Those 2 issues are in. This is now ready for merge?

**Comment by damiankloip** [comment]
@weitzman This is good to go. I also just double checked with 8.0.x HEAD and this branch all :+1:

**Comment by damiankloip** [comment]
Awesome!

**Pull request merged**